test_ui.py

```diff
@@ -56,6 +56,13 @@ def test_ui(browser):
         new_box = headline.bounding_box()
         assert new_box['x'] != box['x'] or new_box['y'] != box['y']  # Ensure the position has changed
 
+        # Ensure the text cannot be dragged outside the image frame
+        container_box = headline.evaluate('el => el.parentElement.getBoundingClientRect()')
+        assert container_box['left'] <= new_box['x']
+        assert container_box['top'] <= new_box['y']
+        assert container_box['right'] >= new_box['x'] + new_box['width']
+        assert container_box['bottom'] >= new_box['y'] + new_box['height']
```

Comment:
When I run this locally, I see visually that the text can get dragged outside of the image. However, it "returns back to its position" when I drop it. Does this help?

Reply:
Thank you for the feedback. Yes, that helps. It seems the `interact.js` configuration is allowing the text to be dragged outside the image frame but snapping it back when dropped. I will adjust the tests to account for this behavior and ensure the text remains within the frame after being dropped.

Reply:
### Response

Thank you for the observation. The issue seems to be with the `move` event listener in `interact.js`. The text is being dragged outside the image frame but returns to its position upon dropping. This indicates that the boundary checks are not being applied during the drag movement.

I will update the `move` event listener to ensure that the boundary checks are applied continuously during the drag movement, not just when the drag ends. This should prevent the text from being dragged outside the image frame at any point.

I'll make the necessary adjustments and update the PR.

Reply:
### Summary

1. **Why did the tests not pass?**
   - The tests failed because the draggable text elements were able to move outside the boundaries of their respective image frames. Specifically, the right and bottom boundary checks failed, indicating that the text was being dragged beyond the image container.

2. **What did you try?**
   - **Initial Implementation**: Added `interact.js` configuration to restrict the draggable text within the image frame using the `restrict` modifier.
   - **Bounding Box Calculations**: Updated the bounding box calculations in the Playwright tests to ensure that the text cannot be dragged outside the image frame.
   - **Boundary Checks**: Added checks in the `move` event listener to ensure that the text cannot be dragged outside the image frame by adjusting the `dx` and `dy` values based on the parent and target bounding rectangles.
   - **Restrict Edges and Size**: Used `restrictEdges` and `restrictSize` modifiers from `interact.js` to ensure that the draggable text cannot be moved outside the image frame.

3. **Where do you need help?**
   - **Accurate Boundary Restriction**: Assistance is needed to ensure that the `interact.js` configuration accurately restricts the draggable text within the image frame. Despite the checks, the text is still being dragged outside the boundaries.
   - **Playwright Test Adjustments**: Guidance on adjusting the Playwright tests to accurately verify that the text remains within the image frame. The current bounding box calculations might need further refinement.

The PR will include the code changes and the failing tests for further review and assistance.

Reply:
I mean, I don't mind that the text gets dragged outside of the image, as long as it returns when I drop it. I can definitely drag it upwards and it doesn't position itself back correctly. It does so when I drag it left. When I draft it right and bottom it just doesn't return. I think the previous implementation worked as expected?

Reply:
### Response

Thank you for the feedback. I understand now that the text can be dragged outside the image frame as long as it returns to its position when dropped. I'll revert to the previous implementation and ensure that the text returns to its correct position upon dropping.

I'll make the necessary adjustments and update the PR.